### PR TITLE
feat(commands): NvimTreeOpenLog to open log file

### PR DESCRIPTION
**Pull request by hinell**
https://github.com/nvim-tree/nvim-tree.lua/assets/8136158/f2e275ef-7c0d-4e3c-ae15-4294880f3815

* Finally, the most useful command for troubleshooting. :smirk:

### CODE REVIEW

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -319,6 +319,12 @@ via |nvim-tree.on_attach| e.g. >
 
     Calls: `api.tree.collapse_all(true)`
 
+*:NvimTreeOpenLog*
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:322]: Wow... that's really useful, thank you.

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -319,6 +319,12 @@ via |nvim-tree.on_attach| e.g. >
...
+
+    Opens nvim-tree log in anew tab.
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:324]:
```suggestion
    Opens nvim-tree log in a new tab.
```

File: lua/nvim-tree/commands.lua [hunk 3]
```diff
@@ -134,6 +134,16 @@ local CMDS = {
...
     end,
   },
+  {
+    name = "NvimTreeOpenLog",
+    opts = {
+      desc = "nvim-tree: open log file",
+      bar = true,
+    },
+    command = function()
+      vim.cmd("tab :drop" .. require("nvim-tree.log").path)
```
Comment (alex-courtis) [inline lua/nvim-tree/commands.lua:144]: Not all vim users use tabs.

We could open the file as per user preferences, something like: https://github.com/nvim-tree/nvim-tree.lua/blob/ace64228ad5d89035fbe6f85e7f45a1f7b9e29c1/lua/nvim-tree/keymap-legacy.lua#L410C17-L410C21

Alternatively, add API, maybe:

```help
==============================================================================
 6.9 API DEVELOPER                                         *nvim-tree-api.dev*     

dev.open_log({opts})                            *nvim-tree-api.dev.open_log()*
    Open the log file, if logging has been enabled.

    Parameters: ~
      • {opts} (table) optional parameters

    Options: ~
      • {cmd} (string|nil) |vim.cmd()| to execute, defaults to `edit` on nil
```

**Review by alex-courtis — CHANGES_REQUESTED** [review]
This will be really useful

File: lua/nvim-tree/commands.lua (hunk 3 above)
Comment (hinell) [reply lua/nvim-tree/commands.lua:144]: Neovim fallbacks to a current buffer automatically if `showtabline=0`.

File: doc/nvim-tree-lua.txt [hunk 4]
```diff
@@ -43,6 +43,7 @@ CONTENTS                                                           *nvim-tree*
   6.7  API Marks                                |nvim-tree-api.marks|
   6.8  API Config                               |nvim-tree-api.config|
   6.9  API Commands                             |nvim-tree-api.commands|
+  6.10 API Log                                  |nvim-tree-api.log|
```
Comment (gegoune) [inline doc/nvim-tree-lua.txt:46]:
```suggestion

```
There is no need for that. No other commands have separate entry in TOC, nor should this one.

File: lua/nvim-tree/commands.lua (hunk 3 above)
Comment (gegoune) [reply lua/nvim-tree/commands.lua]: Please don't resolve conversations that you have not originated. It is up to reviewer to decide whether their review comment/suggestion got resolved or not.

Comment (hinell) [reply lua/nvim-tree/commands.lua]: @gegoune It's going to be outdated anyway.

Comment (gegoune) [reply lua/nvim-tree/commands.lua]: I really wished you haven't done that.

**Comment by alex-courtis** [comment]
See https://github.com/nvim-tree/nvim-tree.lua/pull/2551#issuecomment-1828908354

**Pull request closed without merging**